src/lock/postgres.rs

```diff
@@ -0,0 +1,167 @@
+use std::collections::hash_map::DefaultHasher;
+use std::hash::{Hash, Hasher};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::Result;
+use tokio::sync::Mutex;
+use tokio_postgres::{Client, NoTls};
+use tracing::{debug, warn};
+
+use super::DistributedLock;
+
+/// PostgreSQL-based distributed lock using advisory locks.
+///
+/// This lock uses PostgreSQL advisory locks to coordinate access across multiple hosts.
+/// The lock is identified by a 64-bit integer derived from hashing the target path.
+/// Uses tokio-postgres for lightweight connection management.
+pub struct PostgresLock {
+    client: Arc<Mutex<Client>>,
+    lock_key: i64,
+    target_path: PathBuf,
+    owner: Option<String>,
+}
+
+impl PostgresLock {
+    /// Create a new PostgreSQL client connection.
+    /// Uses standard PG* environment variables for connection parameters.
+    async fn connect() -> Result<Arc<Mutex<Client>>> {
+        // Build connection string from environment variables
+        let host = std::env::var("PGHOST").unwrap_or_else(|_| "localhost".to_string());
+        let port = std::env::var("PGPORT").unwrap_or_else(|_| "5432".to_string());
+        let user = std::env::var("PGUSER").unwrap_or_else(|_| "postgres".to_string());
+        let dbname = std::env::var("PGDATABASE").unwrap_or_else(|_| "postgres".to_string());
+        let password = std::env::var("PGPASSWORD").ok();
+
+        let mut config = format!("host={} port={} user={} dbname={}", host, port, user, dbname);
+        if let Some(pwd) = password {
+            config.push_str(&format!(" password={}", pwd));
+        }
+
+        let (client, connection) = tokio_postgres::connect(&config, NoTls).await?;
```

Comment:
NoTls should be a config to pass in, or from env. could you follow the style of this repo, how they manage configuration?